apps/studio/components/interfaces/Storage/StorageExplorer/MoveItemsModal.tsx

```diff
@@ -1,90 +1,127 @@
+  const { selectedItemsToMove, selectedBucket, moveFiles, setSelectedItemsToMove } =
+    useStorageExplorerStateSnapshot()
+  const form = useForm<z.infer<typeof PathSchema>>({
+    resolver: zodResolver(PathSchema),
+    defaultValues: { path: '' },
+    mode: 'onSubmit',
+  })
+
+  const isMoving = form.formState.isSubmitting
 
   const multipleFiles = selectedItemsToMove.length > 1
 
   const title = multipleFiles
-    ? `Moving ${selectedItemsToMove.length} items within ${bucketName}`
+    ? `Moving ${selectedItemsToMove.length} items within ${selectedBucket.name}`
     : selectedItemsToMove.length === 1
-      ? `Moving ${selectedItemsToMove[0]?.name} within ${bucketName}`
+      ? `Moving ${selectedItemsToMove[0]?.name} within ${selectedBucket.name}`
       : ``
 
   const description = `Enter the path to where you'd like to move the file${
     multipleFiles ? 's' : ''
   } to.`
 
-  const onConfirmMove = (event: any) => {
-    if (event) {
-      event.preventDefault()
-    }
-    setMoving(true)
-    const formattedPath = newPath[0] === '/' ? newPath.slice(1) : newPath
-    onSelectMove(formattedPath)
+  const handleClose = () => {
+    setSelectedItemsToMove([])
+    form.reset()
+    onClose?.()
+  }
+
+  const onSubmit: SubmitHandler<z.infer<typeof PathSchema>> = async (values) => {
+    const formattedPath = values.path[0] === '/' ? values.path.slice(1) : values.path
+    await moveFiles(formattedPath)
+    handleClose()
   }
 
   return (
-    <Modal
-      visible={visible}
-      header={title}
-      description={description}
-      size="medium"
-      onCancel={onSelectCancel}
-      customFooter={
-        <div className="flex items-center gap-2">
-          <Button type="default" onClick={onSelectCancel}>
+    <Dialog
+      open={selectedItemsToMove.length > 0}
+      onOpenChange={(open) => {
+        if (!open) {
+          handleClose()
+        }
+      }}
+    >
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>{title}</DialogTitle>
+          <DialogDescription>{description}</DialogDescription>
+        </DialogHeader>
+        <DialogSectionSeparator />
+        <DialogSection>
+          <Form_Shadcn_ {...form}>
+            <form
+              id={formId}
+              className="flex flex-col gap-4"
+              onSubmit={form.handleSubmit(onSubmit)}
+            >
+              <FormField_Shadcn_
+                key="path"
+                name="path"
+                control={form.control}
+                render={({ field }) => (
+                  <FormItemLayout
+                    name="path"
+                    label={`Path to new directory in ${selectedBucket.name}`}
+                    description="Leave blank to move items to the root of the bucket"
+                  >
+                    <FormControl_Shadcn_>
+                      <Input_Shadcn_
+                        id="path"
+                        {...field}
+                        autoFocus
```

Comment:
so far this is the only modal I've come across that had `autoFocus` set on an input. I'm not sure we want to be doing this as it could interfere with screen readers